Both console and assert e2e IT  failed.
`org.apache.seatunnel.e2e.flink.fake.FakeSourceToConsoleIT`
```
22/06/30 17:18:28 ERROR FlinkContainer: Error: Could not find or load main class org.apache.seatunnel.core.flink.FlinkEnvParameterParser


java.lang.AssertionError: 
Expected :0
Actual   :1
<Click to see difference>

```

--- a/seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/batch/FlinkBatchExecution.java
+++ b/seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/batch/FlinkBatchExecution.java
@@ -118,6 +118,6 @@ public class FlinkBatchExecution implements Execution<FlinkBatchSource, FlinkBat
     }
 
     private boolean whetherExecute(List<FlinkBatchSink> sinks) {
-        return sinks.stream().anyMatch(s -> !"ConsoleSink".equals(s.getPluginName()));
+        return sinks.stream().noneMatch(s -> "ConsoleSink".equals(s.getPluginName()) || "AssertSink".equals(s.getPluginName()));
     }
 }

--- a/seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-assert/src/main/java/org/apache/seatunnel/flink/assertion/sink/AssertSink.java
+++ b/seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-assert/src/main/java/org/apache/seatunnel/flink/assertion/sink/AssertSink.java
@@ -29,6 +29,7 @@ import org.apache.seatunnel.flink.stream.FlinkStreamSink;
 import org.apache.seatunnel.shade.com.typesafe.config.Config;
 
 import com.google.auto.service.AutoService;
+import lombok.SneakyThrows;
 import org.apache.commons.collections.CollectionUtils;
 import org.apache.flink.api.java.DataSet;
 import org.apache.flink.streaming.api.datastream.DataStream;
@@ -50,6 +51,7 @@ public class AssertSink implements FlinkBatchSink, FlinkStreamSink {
     private Config config;
     private List<? extends Config> configList;
 
+    @SneakyThrows
     @Override
     public void outputBatch(FlinkEnvironment env, DataSet<Row> inDataSet) {
         inDataSet.map(row -> {
@@ -59,7 +61,7 @@ public class AssertSink implements FlinkBatchSink, FlinkStreamSink {
                     throw new IllegalStateException("row :" + row + " fail rule: " + failRule);
                 });
             return null;
-        });
+        }).print();
     }
 
     @Override
@@ -71,7 +73,7 @@ public class AssertSink implements FlinkBatchSink, FlinkStreamSink {
                     throw new IllegalStateException("row :" + row + "field name of the fail rule: " + failRule.getFieldName());
                 });
             return null;
-        });
+        }).print();
     }
 
     @Override

--- a/seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-assert/src/main/java/org/apache/seatunnel/flink/assertion/sink/AssertSink.java
+++ b/seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-assert/src/main/java/org/apache/seatunnel/flink/assertion/sink/AssertSink.java
@@ -38,11 +38,11 @@ import org.apache.flink.types.Row;
 import java.util.List;
 
 /**
- * A flink sink plugin which can assert illegal data by user defined rules
- * Refer to https://github.com/apache/incubator-seatunnel/issues/1912
+ * A flink sink plugin which can assert illegal data by user defined rules Refer to https://github.com/apache/incubator-seatunnel/issues/1912
  */
 @AutoService(BaseFlinkSink.class)
 public class AssertSink implements FlinkBatchSink, FlinkStreamSink {
+
     //The assertion executor
     private static final AssertExecutor ASSERT_EXECUTOR = new AssertExecutor();
     //User defined rules used to assert illegal data
@@ -54,14 +54,16 @@ public class AssertSink implements FlinkBatchSink, FlinkStreamSink {
     @SneakyThrows
     @Override
     public void outputBatch(FlinkEnvironment env, DataSet<Row> inDataSet) {
-        inDataSet.map(row -> {
-            ASSERT_EXECUTOR
-                .fail(row, assertFieldRules)
-                .ifPresent(failRule -> {
-                    throw new IllegalStateException("row :" + row + " fail rule: " + failRule);
-                });
-            return null;
-        }).print();
+        try {
+            inDataSet.collect().forEach(row ->
+                ASSERT_EXECUTOR
+                    .fail(row, assertFieldRules)
+                    .ifPresent(failRule -> {
+                        throw new IllegalStateException("row :" + row + " fail rule: " + failRule);
+                    }));
+        } catch (Exception ex) {
+            throw new RuntimeException("AssertSink execute failed", ex);
+        }
     }
 
     @Override
@@ -73,7 +75,7 @@ public class AssertSink implements FlinkBatchSink, FlinkStreamSink {
                     throw new IllegalStateException("row :" + row + "field name of the fail rule: " + failRule.getFieldName());
                 });
             return null;
-        }).print();
+        });
     }
 
     @Override

